Jumpscares/Enemy spawn events added

Mattarr opened the pull request:
+Added trigger volumes to strategic locations in stage to coincide with player movement and cause jumpscares.
+Added Invisible wall + text indicator in market area to prevent skipping jumpscare event.
+Added 'Rats' - currently small patrol meshes with increased speed.
+Modified the "Trigger" script (Added "ModifiedTrigger" file) to only check for the collider of a "Player" - previously it would trigger if any object with a collider entered the volume, which could potentially break/lessen the impact of jumpscare events.

TODO:
-Replace 'Rats' with rat meshes/animations
-Patrolling guards need a bit of overhaul with their vision/detection, running perpendicularly to their field of view can sometimes cause them to not lock on and initiate the attack script, even at close quarters.
-Lock scripts need to be re-enabled to test the triggers that spawn the 'rats' at the entrance to the sewers.

Closes #243

poetahto commented:
I consolidated the changes to the triggers so it's all in one Trigger script now - should remove any confusion about the differences between them. I also added a "one shot" flag to the Trigger script, which will automatically disable them after being activated once. This should make future work in this area easier.

Also, is this invisible wall supposed to be disabled at some point? I think we should probably clarify that before shipping the playtest build, as a lot of people might mistake it as a bug. 

![wall](https://user-images.githubusercontent.com/11669335/159373578-756b7327-1bd4-4988-bd1b-c46a7a63b8cb.png)

Besides that, it looks good! I would add that using Timeline to orchestrate things like the rat animation is probably what we want to transition to in the future, but for now the current solution works.

Mattarr commented:
(Don't know why I wasn't given the prompt for a pull request).
Made small rework as discussed with @alyg563:
+Moved first enemy to other side of corridor
-Removed invisible wall and text in side of market area
+Added enemy spawn near well wheel to push player forward into central hub (enemy spawns with first jumpscare trigger)
+Added trigger in central hub to despawn well wheel enemy

MINOR BUG
The "spikes" on the archways appear to have collision and can potentially block movement if the player doesn't crouch.
![image](https://user-images.githubusercontent.com/92897307/159594634-1bbe98e2-a254-47a5-ba6f-bd33ceae490a.png)

alyg563 reviewed (approved):
Looks good now!
